### PR TITLE
vibe.web.web: Allow to return strings

### PR DESCRIPTION
**Pull request by wilzbach**
Split-off from #1697

### CODE REVIEW

File: web/vibe/web/web.d [hunk 1]
```diff
@@ -964,6 +964,8 @@ private void handleRequest(string M, alias overload, C, ERROR...)(HTTPServerRequ
 				} else {
 					res.writeBody(ret);
 				}
+			} else static if (is(RET : string)) {
+				res.writeBody(ret);
 			} else {
 				static assert(is(RET == void), M~": Only InputStream, Json and void are supported as return types for route methods.");
```
Comment (WebFreak001) [inline web/vibe/web/web.d]: Don't forget to include string in the error message

**Comment by wilzbach** [comment]
> Don't forget to include string in the error message

Thanks. Added!

**Comment by wilzbach** [comment]
@s-ludwig anything blocking this? Returning a `string` happens often enough, s.t. imho it justifies the convenience and also this is just an added bonus as a couple of return types (`InputStream` and `Json` are already accepted).

**Comment by wilzbach** [comment]
Ping @s-ludwig - I think this is a trivial, non-controversial addition and helps a lot of people as returning `string` is a frequent use case.

**Comment by WebFreak001** [comment]
@s-ludwig merge pls, this is really useful and further changes can be done afterwards

File: web/vibe/web/web.d [hunk 2]
```diff
@@ -1020,8 +1029,10 @@ private void handleRequest(string M, alias overload, C, ERROR...)(HTTPServerRequ
 				} else {
 					res.writeBody(ret);
 				}
+			} else static if (is(RET : string)) {
```
Comment (s-ludwig) [inline web/vibe/web/web.d]: `const(char)[]` is sufficient here.

File: web/vibe/web/web.d [hunk 3]
```diff
@@ -1020,8 +1029,10 @@ private void handleRequest(string M, alias overload, C, ERROR...)(HTTPServerRequ
...
+				res.writeBody(ret);
 			} else {
-				static assert(is(RET == void), M~": Only InputStream, Json and void are supported as return types for route methods.");
+				static assert(is(RET == void), M~": Only InputStream, Json, string and void are supported as return types for route methods.");
```
Comment (s-ludwig) [inline web/vibe/web/web.d]: Would be nice to add `const(ubyte)[]` here, too (and change to `const(char)[]` if adjusted).

**Comment by s-ludwig** [comment]
Couldn't find a compelling argument against this, especially considering that `ubyte[]` and `char[]` are already supported by `writeBody` in a similar way. Would just be nice to loosen this up to `const(char)[]` instead of `string`.

**Comment by wilzbach** [comment]
> Would just be nice to loosen this up to const(char)[] instead of string.

Good idea. Done.

**Pull request merged**